swap/swap.go

```diff
@@ -78,9 +80,34 @@ func NewParams() *Params {
 	}
 }
 
+// NewLogger returns a new logger
+func NewLogger(h interface{}) log.Logger {
+	logLevel := 3
+	swapLogger := log.New("swaplog", "*")
+	hd, ok := h.(log.Handler)
+
+	if !ok {
+		//default handler just in case the handler is not set
+		hd = log.Root().GetHandler()
+		swapLogger.SetHandler(hd)
+		return swapLogger
+	}
+
+	//Filters messages with the correct logLevel for swap
+	hd = log.LvlFilterHandler(log.Lvl(logLevel), hd)
```

Comment:
not clear on this one—shouldn't `LvlFilterHandler` be called even if `ok == false`?

Reply:
answer: this should only be called for the rotating file handler, so as not to overwrite the filter previously set for the default handler.